include/boost/beast/core/async_base.hpp

```diff
@@ -186,7 +186,22 @@ class async_base
         "Executor type requirements not met");
 
     Handler h_;
-    net::executor_work_guard<Executor1> wg1_;
+    detail::select_work_guard_t<Executor1> wg1_;
+
+public:
+    /** The type of executor associated with this object.
+
+    If a class derived from @ref async_base is a completion
+    handler, then the associated executor of the derived class will
+    be this type.
+*/
+    using executor_type = typename
+        net::associated_executor<
+            Handler,
+            typename detail::select_work_guard_t<Executor1>::executor_type
```

Comment:
This is kind of annoying, a `detail` type will show up in the docs.

Reply:
we could replace it with a __deduced__ in the docs?

Reply:
You might be able to say this instead:
```
#ifdef BOOST_BEAST_DOCS
    using executor_type = typename
        net::associated_executor<
            Handler,
            __implementation_defined__>::type
#else
  ...
#endif
```

Or just

```
using executor_type = __see_below__;
```

Reply:
yeah deduced might work, check the main.qbk for the exact spelling